sgx/src/attestation_types/ti.rs

```diff
@@ -21,6 +22,29 @@ pub struct TargetInfo {
     reserved2: [u64; 25],
 }
 
+#[cfg(feature = "get_report")]
+impl TargetInfo {
+    /// This function takes a 64-byte REPORTDATA structure as an argument
+    /// calls EREPORT
+    /// returns a cryptographic REPORT structure
+    pub fn get_report(&self, data: &[u16; 32]) -> report::Report {
+        #[repr(C, align(128))]
+        struct Data<T>(T);
+        let data = Data(data);
+        let report: report::Report = Default::default();
+        unsafe {
+            asm!(
+            "enclu",
+            in("rax") 00,
+            in("rdx") &report,
+            in("rbx") self,
+            in("rcx") &data,
+            );
+        }
+        report
+    }
+}
```

Comment:
I'm not sure this code does what you think it does. In particular, you're aligning a pointer to a buffer, not the buffer itself. Therefore, I suspect this actually corrupts memory. You probably want this:

```suggestion
#[repr(C, align(128))]
pub ReportData(pub [u8; 64]);

#[cfg(feature = "get_report")]
impl TargetInfo {
    /// Generate a report to the specified target with the included data.
    pub fn get_report(&self, data: &ReportData) -> report::Report {
        const EREPORT: usize = 0;

        unsafe {
            let mut report = core::mem::MaybeUninit::<report::Report>::uninit();

            asm!(
                "enclu",
                in("rax") EREPORT,
                in("rbx") self,
                in("rcx") data.0.as_ptr(),
                in("rdx") report.as_mut_ptr(),
            );

            report.assume_init()
        }
    }
}
```